File: c/hamming/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs) {
    int count = 0;
    int i = 0;
    
    while(lhs[i] && rhs[i]) {
        if (lhs[i] != rhs[i]) {
            count++;
        }

        i++;
    }

    if (lhs[i] || rhs[i]) {
        return -1;
    }
    
    return count;
}

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(seq_a,seq_b)
    if #seq_a ~= #seq_b then
        return -1
    end

    local distance = 0

    for i = 1, #seq_a do
        local a_char = seq_a:sub(i, i)
        local b_char = seq_b:sub(i, i)

        if a_char ~= b_char then
            distance = distance + 1
        end
    end

    return distance
end

return Hamming
